### PR TITLE
Update Model section of developer guide

### LINKED ISSUE #79: Update Model section of Developer Guide

### PR DESCRIPTION
**Pull request by zaidansani**
Resolves #79 

Reflected new AppointmentBook model, as well as changes to existing Person model.

### CODE REVIEW

File: docs/diagrams/ModelClassDiagram.puml
Comment (rxchell) [inline docs/diagrams/ModelClassDiagram.puml:1]: LGTM. 

The diagram appears fine in puml, but the arrows appear different in the docs.

Screenshot of the docs (some arrows are not pointing to their corresponding class)
Screenshot 2024-10-16 at 16 14 31

File: docs/diagrams/BetterModelClassDiagram.puml
Comment (rxchell) [inline docs/diagrams/BetterModelClassDiagram.puml:1]: For the description if the BetterModelClassDiagram under **Note:** (line 140), the "`Tag` list in the `AddressBook`" can be specified to be `UniqueTagList`

**Review by rxchell — CHANGES_REQUESTED** [review]

File: docs/diagrams/ModelClassDiagram.puml
Comment (zaidansani) [reply docs/diagrams/ModelClassDiagram.puml:1]: This is a weird issue, no idea whats the issue

**Review by rxchell — APPROVED** [review]
LGTM!

**Pull request merged**